### ISSUE TITLE
[bug] Wrong image size extracted in extract_features_vmb.py

### ISSUE THREAD

**Issue by ronghanghu** [facebookresearch/pythia#120]
## 🐛 Bug

The image width and image height extracted with extract_features_vmb.py in https://github.com/facebookresearch/pythia/blob/64ac866210241d769a36a02f89d25e5b244d579b/pythia/scripts/features/extract_features_vmb.py#L118-L119 are wrong. The "image_width" and "image_height" keys in the extracted *_info.npy files do not match the actual image size.

This can cause errors if one would like to compute the relative box coordinates from the detected boxes.

### Underlying cause

The root cause of this bug goes back to the vqa-maskrcnn-benchmark repository, where the bug is introduced in commit https://gitlab.com/meetshah1995/vqa-maskrcnn-benchmark/commit/2d636cfabadb4698a61708037ba94cade62fea7a

After the above commit, `to_image_list` in vqa-maskrcnn-benchmark will generate wrong image sizes, as they are set to `image_sizes = [im.shape[-2:] for im in batched_imgs]`, where `batched_imgs` is a padded tensor of the maximum width and height for the batch. Details in https://gitlab.com/meetshah1995/vqa-maskrcnn-benchmark/blob/2d636cfabadb4698a61708037ba94cade62fea7a/maskrcnn_benchmark/structures/image_list.py#L66.

This causes the `image_sizes` in vqa-maskrcnn-benchmark to be the maximum size of all images in a batch, which results in wrong image sizes computed in https://github.com/facebookresearch/pythia/blob/64ac866210241d769a36a02f89d25e5b244d579b/pythia/scripts/features/extract_features_vmb.py#L118-L119.

### Proposed solution

Directly setting "image_width" and "image_height" keys from the loaded images, instead of setting them from the vqa-maskrcnn-benchmark output

## Command

```
python3 pythia/scripts/features/extract_features_vmb.py --image_dir data/images --model_file data/detectron_model.pth --config_file data/detectron_model.yaml
```

## To Reproduce

1. Download the images to data/images
2. Run `python3 pythia/scripts/features/extract_features_vmb.py --image_dir data/images --model_file data/detectron_model.pth --config_file data/detectron_model.yaml`

## Expected behavior

The "image_width" and "image_height" keys in the extracted *_info.npy files should be correct.

## Environment

PyTorch version: 1.2.0.dev20190622
Is debug build: No
CUDA used to build PyTorch: 9.0.176

OS: Ubuntu 18.04.1 LTS
GCC version: (Ubuntu 5.5.0-12ubuntu1) 5.5.0 20171010
CMake version: version 3.9.4

Python version: 3.6
Is CUDA available: Yes
CUDA runtime version: 9.0.176
GPU models and configuration: 
GPU 0: Quadro GP100
GPU 1: Quadro GP100

Nvidia driver version: 410.79
cuDNN version: Could not collect

**Comment by meetshah1995** [comment]
@ronghanghu - Thanks for pointing this out. This bug (or feature) is present to imitate Detectron like behaviour.

Without this change, I observed a difference in features extracted from C2 models (extracted using C2) and features extracted from C2 models converted to Pytorch (extracted using Pytorch). I explicitly added this [line](https://gitlab.com/meetshah1995/vqa-maskrcnn-benchmark/blob/2d636cfabadb4698a61708037ba94cade62fea7a/maskrcnn_benchmark/structures/image_list.py#L66) to make sure the `im_scale` passed to the model for scaling are similar to the `im_scale` sent in Detectron [im blobs](https://github.com/facebookresearch/Detectron/blob/cbb0236dfdc17790658c146837215d2728e6fadd/detectron/utils/blob.py#L100-L125).  

The original maskrcnn-benchmark does the [right thing](https://github.com/facebookresearch/maskrcnn-benchmark/blob/master/maskrcnn_benchmark/structures/image_list.py#L54-L68) by inferring the `im_scale` from the `tensor` shape instead of `batched_imgs`.

**Comment by ronghanghu** [comment]
Thanks for the explanation! I think as a workaround one can directly extracting the image height and width from `_image_transform` in extract_features_vmb.py, instead of recomputing them from the vqa-maskrcnn-benchmark outputs.

**Comment by endernewton** [comment]
@meetshah1995 does the change in features affect pythia performance? I mean if one can either go with the C2 tradition or go with the PT benchmark, but make sure the features extracted are *consistent* for training and testing.

**Comment by meetshah1995** [comment]
Yes, the change in features (if not using the above fix) affects Pythia performance by (-ve) 2.5 in VQA score on test-dev.

**Comment by liiiiiiiiil** [comment]
> ## 🐛 Bug
> The image width and image height extracted with extract_features_vmb.py in
> 
> https://github.com/facebookresearch/pythia/blob/64ac866210241d769a36a02f89d25e5b244d579b/pythia/scripts/features/extract_features_vmb.py#L118-L119
> 
> are wrong. The "image_width" and "image_height" keys in the extracted *_info.npy files do not match the actual image size.
> This can cause errors if one would like to compute the relative box coordinates from the detected boxes.
> 
> ### Underlying cause
> The root cause of this bug goes back to the vqa-maskrcnn-benchmark repository, where the bug is introduced in commit https://gitlab.com/meetshah1995/vqa-maskrcnn-benchmark/commit/2d636cfabadb4698a61708037ba94cade62fea7a
> 
> After the above commit, `to_image_list` in vqa-maskrcnn-benchmark will generate wrong image sizes, as they are set to `image_sizes = [im.shape[-2:] for im in batched_imgs]`, where `batched_imgs` is a padded tensor of the maximum width and height for the batch. Details in https://gitlab.com/meetshah1995/vqa-maskrcnn-benchmark/blob/2d636cfabadb4698a61708037ba94cade62fea7a/maskrcnn_benchmark/structures/image_list.py#L66.
> 
> This causes the `image_sizes` in vqa-maskrcnn-benchmark to be the maximum size of all images in a batch, which results in wrong image sizes computed in
> 
> https://github.com/facebookresearch/pythia/blob/64ac866210241d769a36a02f89d25e5b244d579b/pythia/scripts/features/extract_features_vmb.py#L118-L119
> 
> .
> ### Proposed solution
> Directly setting "image_width" and "image_height" keys from the loaded images, instead of setting them from the vqa-maskrcnn-benchmark output
> 
> ## Command
> ```
> python3 pythia/scripts/features/extract_features_vmb.py --image_dir data/images --model_file data/detectron_model.pth --config_file data/detectron_model.yaml
> ```
> 
> ## To Reproduce
> Steps to reproduce the behavior:
> 
> 1. Download the images to data/images
> 2. Run `python3 pythia/scripts/features/extract_features_vmb.py --image_dir data/images --model_file data/detectron_model.pth --config_file data/detectron_model.yaml`
> 
> ## Expected behavior
> The "image_width" and "image_height" keys in the extracted *_info.npy files should be correct.
> 
> ## Environment
> PyTorch version: 1.2.0.dev20190622
> Is debug build: No
> CUDA used to build PyTorch: 9.0.176
> 
> OS: Ubuntu 18.04.1 LTS
> GCC version: (Ubuntu 5.5.0-12ubuntu1) 5.5.0 20171010
> CMake version: version 3.9.4
> 
> Python version: 3.6
> Is CUDA available: Yes
> CUDA runtime version: 9.0.176
> GPU models and configuration:
> GPU 0: Quadro GP100
> GPU 1: Quadro GP100
> 
> Nvidia driver version: 410.79
> cuDNN version: Could not collect

hi， could you please answer me a question? How did you install the vqa-maskrcnn-benchmark, I check the install.md  that is the same with maskrcnn-benchmark. please!

**Comment by liiiiiiiiil** [comment]
> Yes, the change in features (if not using the above fix) affects Pythia performance by (-ve) 2.5 in VQA score on test-dev.

Could you please tell me how to install the vqa-maskrcnn-benchmark, I check the install.md that is the same with maskrcnn-benchmark. please!
